### PR TITLE
Multiple Authors Access

### PR DESCRIPTION
**Pull request by sorochak**
This PR introduces functionality to allow multiple users to have access to edit the same record. It addresses the feature request outlined in issue https://github.com/cioos-siooc/metadata-entry-form/issues/78, aiming to enhance collaborative efforts and streamline the editing process on shared records. Only the original author of the record has the ability to share editing access with others. When a user logs in that has access to another user's records, they see a "Shared with me" link in the left sidebar that links to a `Shared` records page. They can edit but not submit or delete a record shared with them.

# Changes Implemented:
* Data Structure Modifications: Introduced a `sharedWith` node under each record for listing IDs of users with whom the record is shared. Also added a `shares` node at the regional level to map users to the records they're sharing.
* Security Rule Updates: Revised security rules to support shared editing, ensuring that only authorized users can edit shared records.
* Front End Enhancements:
   * Added a 'Shared with Me' section in the navigation drawer, conditionally rendered for users with shared records.
   * Implemented UI changes to allow users to share records by entering email addresses, mirroring the approach used in the contacts tab for consistency and user-friendliness.

### CODE REVIEW

**Comment by JessyBarrette** [comment]
That's a nice add-on @sorochak. I'm wondering if it would be better to use suggestions based on what the user is typing instead a providing a dropdown of our whole list of user emails.

Maybe, it would also be better to use names (name/lastname) instead as a list/suggestion.

I'm more worried about peoples email privacy than anything.

File: firebase-functions/functions/datacite.js [hunk 1]
```diff
@@ -184,3 +185,567 @@ exports.getDoiStatus = functions.https.onCall(async (data) => {
...
+
+    // Check for non-null and non-empty
+    return authHash && authHash !== "" && prefix && prefix !== "";
+  } catch (error) {
+    console.error("Error checking Datacite credentials:", error);
+    return false;
+  }
+});
+
+exports.recordToDataCite = functions.https.onCall(async (data) => {
```
Comment (fostermh) [inline firebase-functions/functions/datacite.js:436]: the datacite prefix isn't sensitive informaton so recordToDataCite dosn't need to be a firebase function. In order to keep it consistent with the other templates I Think it's better for this to be in the front end with the other templates files as it was.

File: firebase-functions/functions/datacite.js [hunk 2]
```diff
@@ -184,3 +185,567 @@ exports.getDoiStatus = functions.https.onCall(async (data) => {
...
+
+exports.createDraftDoiPR78 = functions.https.onCall(async (data) => {
+
+  const { record, region } = data;
+
+  let prefix;
+  let authHash
+
+  try {
+    prefix = (await admin.database().ref('admin').child(region).child("dataciteCredentials").child("prefix").once("value")).val();
```
Comment (fostermh) [inline firebase-functions/functions/datacite.js:198]: why not use getCredentialsStored here so you don't have to update db paths in multiple places?

**Review by fostermh — CHANGES_REQUESTED** [review]

**Comment by sorochak** [comment]
Thank you for he input @JessyBarrette! It's now a searchable list of names.

**Review by fostermh — APPROVED** [review]

**Pull request merged**